src/main/java/Application.java

```diff
@@ -1,21 +1,119 @@
         fileManager = new FileManager();
         taskManager = new TaskManager();
+        sc = new Scanner(System.in);
+
         dataDirectoryName = "../../../data/";
         dataFileName = "data.txt";
     }
 
+    public void runApplication() {
+        UserInterface.greetUser();
+        String[] userInput = sc.nextLine().split(" ", 2);
+        String command = userInput[0];
+
+        while (!command.equals("bye")) {
+            try {
+                checkUserInput(userInput);
+                if (command.equals("list")) {
+                    taskManager.printTaskList(taskManager.getTaskList());
+                }
+                else if (command.equals("done")){
+                    Task currentTask = taskManager.getTask(Integer.parseInt(userInput[1]));
+                    currentTask.setDone();
+                    System.out.println(UserInterface.line + "\tGood job! You've just completed this task:\n" + "\t\t" + currentTask + "\n" + UserInterface.line);
+                }
+                else if (command.equals("delete")){
+                    int taskNumber = Integer.parseInt(userInput[1]);
+                    Task deletedTask = taskManager.getTask(taskNumber);
+                    taskManager.deleteTask(taskNumber);
+                    System.out.println(UserInterface.line + "\tGot it! I've removed this task for you:\n\t\t" + deletedTask + "\n" + "\tYou now have " + taskManager.getNumOfTasks() + " tasks in total.\n" + UserInterface.line);
+                }
+                else {
+                    Task newTask = null;
+
+                    switch (command) {
+                    case "todo":
+                        newTask = new Todo(userInput[1]);
+                        break;
+                    case "deadline": {
+                        String[] deadlineParams = userInput[1].split(" /");
+                        newTask = new Deadline(deadlineParams[0], deadlineParams[1]);
+                        break;
+                    }
+                    case "event": {
+                        String[] deadlineParams = userInput[1].split(" /");
+                        newTask = new Event(deadlineParams[0], deadlineParams[1]);
+                        break;
+                    }
+                    }
+                    taskManager.addTask(newTask);
+                    System.out.println(UserInterface.line + "\tOne more task added to the hustle:\n\t\t" + newTask + "\n" + "\tYou now have " + taskManager.getNumOfTasks() + " tasks in total.\n" + UserInterface.line);
+                }
+
+            } catch (ToDoBeastException e) {
+                System.out.println(UserInterface.line + "\t" + e.getMessage() + "\n" + UserInterface.line);
+            } finally {
+                userInput = sc.nextLine().split(" ", 2);
+                command = userInput[0];
+            }
+        }
+        UserInterface.exitDisplay();
+    }
+
+    public void checkUserInput(String[] userInput) throws ToDoBeastException {
```

Comment:
Maybe can consider breaking up into smaller functions to help parse the input and check if the input is valid?